### PR TITLE
[TSI] Create hierarchy client and implement list hierarchies

### PR DESCRIPTION
**Pull request by bikamani**
Implementation of List Hierarchies.

### CODE REVIEW

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    /// <summary>
+    /// Hierarchies client that can be used to perform operations such as creating, listing, replacing and deleting Time Series hierarchies.
+    /// </summary>
+    public class HierarchiesClient
+    {
+        private readonly TimeSeriesHierarchiesRestClient _hierarchiesRestClient;
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        /// <summary>
+        /// Initializes a new instance of ModelSettings. This constructor should only be used for mocking purposes.
```
Comment (barustum) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:21]: HierarchiesClient

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            _hierarchiesRestClient = hierarchiesRestClient;
+            _clientDiagnostics = clientDiagnostics;
+        }
+
+        /// <summary>
+        /// Gets Time Series Insights hierarchies in pages asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="AsyncPageable{TimeSeriesHierarchy}"/> of Time Series hierarchies with the http response.</returns>
+        public virtual AsyncPageable<TimeSeriesHierarchy> GetHierarchiesAsync(
```
Comment (barustum) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:41]: GetAsync as opposed to GetHierarchiesAsync.

Comment (bikamani) [reply sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:41]: yes that's correct. Let me change that.

**Comment by barustum** [comment]
Not sure if you plan to do it in this PR, but you also need to createa a new Hierarchies property in TimeSeriesInsightsClient.cs.

**Review by barustum — APPROVED** [review]

**Comment by bikamani** [comment]
> Not sure if you plan to do it in this PR, but you also need to createa a new Hierarchies property in TimeSeriesInsightsClient.cs.

will do.

**Pull request merged**